File: task-03/Haskell/prime_numbers_upto_n.hs
isPrime :: Int -> Bool
isPrime n = null [x | x <- [2..(n-1)], n `mod` x == 0]

primes :: Int -> [Int]
primes n = [x | x <- [2..n], isPrime x]

main :: IO ()
main = do
    putStrLn "Please enter a number:"
    n <- readLn
    let primeNumbers = primes n
    putStrLn $ "Prime numbers up to " ++ show n ++ " are: " ++ show primeNumbers

File: task-03/Python/prime_numbers_up_to_n.py
t=input("Enter a number (n): ")

if t.isdigit():
    n=int(t)

    if n < 2:
        print("There are no prime numbers less than 2.")
    else:
        prime_numbers = []

        for num in range(2, n + 1):
            is_prime = True
            for divisor in range(2, int(num**0.5) + 1):
                if num % divisor == 0:
                    is_prime = False
                    break
            if is_prime:
                prime_numbers.append(num)

        print("Prime numbers up to", n, "are:", prime_numbers)
else:
    print("Invalid input. Please enter a valid number")
